### ISSUE TITLE
[Enhancement]: AppFlowy Serice

### ISSUE THREAD

**Issue by imaron85** [coollabsio/coolify#4458]
### Request Type

New Service

### Description

I would like a Coolify Service for easily deploying AppFlowy.
I've tried myself, but can't figure out recreating the nginx config in coolify.
It might not even be neccessary, all I want is be able to easily deploy AppFlowy for our small team.

**Comment by LeoMartinDev** [comment]
Any news on that ?

**Comment by laGameTV** [comment]
bump as of the release of https://github.com/AppFlowy-IO/AppFlowy-Web

**Comment by TheZomb** [comment]
Is there any news to this?

**Comment by imaron85** [comment]
Made the bounty a nice round number again, we are desperate, lol

**Comment by danabiu** [comment]
any news on this?

**Comment by gleatd01** [comment]
> any news on this?

I'll be working on it this week, or are you working on it?

**Comment by danabiu** [comment]
> > any news on this?
> 
> I'll be working on it this week, or are you working on it?

im still working on but i got stuck, if you want we can work together, i can give you my code with the progress i made.

**Comment by imaron85** [comment]
Any updates?

**Comment by TheZomb** [comment]
@Aditya-Choudhry yes it seems the others aren't working on it anymore

**Comment by Aditya-Choudhry** [comment]
/urgent sir can you help me i need information to work #4458

**Comment by imaron85** [comment]
> /urgent sir can you help me i need information to work [#4458](https://github.com/coollabsio/coolify/issues/4458)

Hi there, I'm not sure what information is missing. This is just about adding AppFlowy as a service to coolify, meaning essentially just creating a compose file that works out of the box with coolify.

**Comment by laGameTV** [comment]
Any progress on this?

**Comment by Aditya-Choudhry** [comment]
/reply i am trying but unable to do it sorry i need help

**Comment by laGameTV** [comment]
bump

**Comment by neooriginal** [comment]
bump

**Comment by MMTE** [comment]
Hi. I have done this, and I will submit the PR today.

**Comment by neooriginal** [comment]
> Hi. I have done this, and I will submit the PR today.

would you able to link me to your repository for it? need it asap for my project

**Comment by MMTE** [comment]
> would you able to link me to your repository for it? need it asap for my project
Here’s a refined version for GitHub:

Check this out—I hope it’s helpful and works for your project.  

**Current Status:**  
- It's a bit tricky to register and receive emails at the moment. I'm still configuring things and need to make a few changes.  
- If you can connect an SMTP server with TLS, you can do that from the **environments section**.  
- Alternatively, you can set a password for the user directly in the database and log in via `/console`.  


maybe If you set the environment to `local` or something non-production, the `admin@example.com` account might be created automatically, making it easier to log in.  

📄 [Here’s the template](https://gist.github.com/MMTE/a76be05c17bff26850e68339717914f2)
1. Create a new resource → Select **Docker Compose (empty)**  
2. Paste the template and run it  

- All services seem fine, except the `ai` service, which requires an **OpenAI API key**.  
- I’m using **SMTP4DEV** for testing emails—it works, but I haven’t exposed its domain or server to Traefik yet.  

I’ll make some enhancements soon and submit a PR. Let me know if this works for now! 😊

**Comment by neooriginal** [comment]
Can not login / create account sadly

**Comment by MMTE** [comment]
> Can not login / create account sadly

hope I can complete that today . I would create a PR when it's complete.

**Comment by neooriginal** [comment]
> > Can not login / create account sadly
> 
> hope I can complete that today . I would create a PR when it's complete.

how is it going?

**Comment by MMTE** [comment]
> how is it going?

[I've updated that.
](https://gist.github.com/MMTE/a76be05c17bff26850e68339717914f2)

- Create a new resource → Select Docker Compose (empty)
- Paste the template and run it
- you can access the email area at https://[your-coolify-appflowy]-address/smtp4dev 
- there you can confirm incoming emails.
- for production obviously you need to replace it with a secure smtp mail server or service.   
- I haven't checked other scenarios like mobile or desktop app yet so I can't confirm if it's complete or not.

**Comment by neooriginal** [comment]
> > how is it going?
> 
> [I've updated that. ](https://gist.github.com/MMTE/a76be05c17bff26850e68339717914f2)
> 
> * Create a new resource → Select Docker Compose (empty)
> * Paste the template and run it
> * you can access the email area at https://[your-coolify-appflowy]-address/smtp4dev
> * there you can confirm incoming emails.
> * for production obviously you need to replace it with a secure smtp mail server or service.
> * I haven't checked other scenarios like mobile or desktop app yet so I can't confirm if it's complete or not.

works nicely, but changing the domain breaks it

**Comment by MMTE** [comment]
> works nicely, but changing the domain breaks it

thanks for mentioning. I would investigate that.

**Comment by MMTE** [comment]
> works nicely, but changing the domain breaks it

I checked that and I was able to change the domain.

- at the nginx services you can change the domain but keep the port which is 80. 
- then restart (or redeploy) to apply the new configuration.
- I believe thee new domain would be accessible after restart. 

![Image](https://github.com/user-attachments/assets/2c037f93-b894-4385-bf37-b378d064d00f)

**Comment by neooriginal** [comment]
It is accessible, but when logging in it still uses the old domain (I saw via console)

-------- Ursprüngliche Nachricht --------
Am 04.04.25 07:50 um Mahdi Taleghani schrieb :

**Comment by MMTE** [comment]
right. I appreciate any help. I believe it's from environment variables. maybe FQDN service domain should be implemented in another way.

**Comment by JW148** [comment]
Hi @MMTE, thanks for you efforts in putting the compose file together. I have it running on my Coolify setup but I have a couple of questions:

1. How do I access the AppFlowy Cloud admin panel (as seen below)?

![Image](https://github.com/user-attachments/assets/a6e597e9-b282-411f-904b-c835dbb3c6a0)

2. Can I setup an external S3 storage provider?

3. How can I configure the auth providers?

**Comment by MMTE** [comment]
> Hi @MMTE

Hi, first, I believe you should install the desktop client then it would automatically open the app on your desktop. Otherwise if you want to use web version then I believe the default generated url is the web version itself, the image you send is admin area as I know. 

For the second I have to check that. Maybe s3 environment variables should be available to coolify panel so we can put external s3. I will check that I would back to you. 

For the third it must be handled by environment variables. Again I would mention if I did move forward.

**Comment by JW148** [comment]
> > Hi [@MMTE](https://github.com/MMTE)
> 
> Hi, first, I believe you should install the desktop client then it would automatically open the app on your desktop. Otherwise if 

Thanks for getting back to me @MMTE.

Just to clarify a couple of points on my end:

I have the desktop AppFlowy app and have set the cloud server url to my self-hosted AppFlowy cloud instance. 

The screenshot I provided was from the documentation and I have not been able to access it.

**Comment by lew1s** [comment]
any update on this?

**Comment by MMTE** [comment]
> any update on this?

did a new PR #6158
You can test the Docker Compose code in the PR on Coolify until the PR result is finalized.

**Comment by MMTE** [comment]
it seem Appflowy itself has wrote self-hosting guide for Coolify.
[How to install AppFlowy Cloud using Coolify
](https://appflowy.com/docs/How-to-install-AppFlowyCloud-using-Coolify)
